src/shared/injector.ts

```diff
@@ -1,22 +1,38 @@
-/**
- * @typedef Provider
- * @prop {unknown} [value] - The value for the object
- * @prop {Function} [class] - A class that should be instantiated to create the object
- * @prop {Function} [factory] - Function that should be called to create the object
- */
+type ValueProvider = { value: unknown };
 
-/**
- * @param {Provider} provider
- */
-function isValidProvider(provider) {
+type WithConstructor = {
```

Comment:
Could alternately have called this `Constructible`. The current name is clear enough though.

Reply:
I tried to come up with the `Constructible` word, but my English skills were not there yet 😂 

I like it better. Will change it.